README.md

```diff
@@ -1,9 +1,139 @@
+This specification does not define concrete PerformanceEntry objects.
+Examples of specifications that define new concrete types of PerformanceEntry objects are [Paint Timing](https://github.com/w3c/paint-timing), [User Timing](https://github.com/w3c/user-timing), [Resource Timing](https://github.com/w3c/resource-timing), and [Navigation Timing](https://github.com/w3c/navigation-timing/).
+
+### Performance getters
+
+The [Performance](https://w3c.github.io/hr-time/#sec-performance) interface is augmented with three new methods that can return a list of PerformanceEntry objects:
+
+* `getEntries()`: returns all of the entries available to the [Performance](https://w3c.github.io/hr-time/#sec-performance) object.
+* `getEntriesByType(type)`: returns all of the entries available to the [Performance](https://w3c.github.io/hr-time/#sec-performance) object whose `entryType` matches *type*.
+* `getEntriesByName(name, type)`: returns all of the entries available to the [Performance](https://w3c.github.io/hr-time/#sec-performance) object whose `name` matches *name*.
+If the optional parameter *type* is specified, it only returns entries whose `entryType` matches *type*.
+
+#### Using the getters in JavaScript
+
+The following example shows how `getEntriesByName()` could be used to obtain the first paint information:
+
+```javascript
+// Returns the FirstContentfulPaint entry, or null if it does not exist.
+function getFirstContentfulPaint() {
+  // We want the entry whose name is "first-contentful-paint" and whose entryType is "paint".
+  // The getter methods all return arrays of entries.
+  const list = performance.getEntriesByName("first-contentful-paint", "paint");
+  // If we found the entry, then our list should actually be of length 1,
+  // so return the first entry in the list.
+  if (list.length > 0)
+    return list[0];
+  // Otherwise, the entry is not there, so return null.
+  else
+    return null;
+}
+```
+
+### PerformanceObserver
+
+A PerformanceObserver object can notified of new PerformanceEntry objects, according to their `entryType` value.
+The construction of the object must receive a callback, which will be ran whenever the user agent is dispatching new entries whose `entryType` value match one of the ones being observed by the observer.
+This callback is not run once per PerformanceEntry nor immediately upon creation of a PerformanceEntry.
+Instead, entries are 'queued' at the PerformanceObserver, and the user agent can execute the callback later.
+When the callback is executed, all queued entries are passed onto the function, and the queue for the PerformanceObserver is reset.
+The PerformanceObserver initially does not observer anything: the `observe()` method must be called to specify what kind of PerformanceEntry objects are to be observed.
+There are two ways to call the `observe()` method, and those should not be mixed within the same PerformanceObserver (an exception will be thrown if this is attempted).
```

Comment:
That sentence is not very clear. Maybe "The `observe()` method has two different modes where it can be called with either an entryTypes array or with a single type, as detailed below. Those modes cannot be mixed, or an exception will be thrown"?